### DISCUSSION TITLE
What parameters are recommended for buildings

### DISCUSSION THREAD

**Discussion by adam-bielanski** [alicevision/Meshroom#1324] [Q&A]
I'll return with (almost) the same question as posted before - did anyone made a "path"of Meshroom parameters that would avoid "skin rash" on completely flat surface, but would preserve sharp edges where needed?

**Comment by natowi** [comment]
I think this is quite dataset and hardware setup specific. Please describe your project in more detail (what camera, lighting?, what kind of "flat surfaces"? reflective? feature less?...)

**Reply by adam-bielanski** [reply]
I think the atttached image explains what i meant. I generally intend to test it for architectural objects with walls, collumns, floors that are relatively "flat", but Meshroom always (at least for my little knowledge of the programme) provides me a very dense mesh that makes bumpy surfaces and "soft" rounded edges. I realise it can  be improved by MeshDecimate and other mesh post processing, but always it gives even more rounded edges and is even less usefull than pure mesh.

So, returning to basic question - is there any tested set of settings for every step to achieve model that is sharp, where it shoud be sharp, and flat on flat surfaces.
I've tested it on: a)sample image set from the net, b) set of photos from my phone camera (Redmi Note 9), c) DSLR Nikon camera

![github](https://user-images.githubusercontent.com/79451289/110684693-bc30c380-81dd-11eb-9b32-a27946f952df.jpg)

**Reply by natowi** [reply]
> architectural objects with walls, columns, floors

the reconstructions will most likely be better that with your test notebook scan.
Most building walls tend to have a good surface structure (setting aside modern architecture with lots of glass and glossy metal).
As you can see in the images of a reconstructed key posted here https://github.com/alicevision/meshroom/issues/566 it is possible to get nice edges. But this really depends on your actual capturing setup and object of interest. I´d recommend you to create a simple dataset for your use case. This is a good example: https://github.com/alicevision/meshroom/issues/1306
**(ignore the annotations)**
![draft21](https://user-images.githubusercontent.com/8138537/109703056-52893780-7b95-11eb-8e7a-633b9be621b9.jpg)
As you can see, the edges are not perfectly sharp, but sharp enough - but that´s the draft meshing result. With depthmap it is more accurate. 
You can tweak the settings in the Meshing and MeshFiltering node to improve the model and reduce the filtering intensity.
Duplicate the nodes and try different iterations to find the best settings for your use case.

**Reply by adam-bielanski** [reply]
Well.... I think generally I understand it, but I have two major issues:
1.  As a complete dummie for coding and mesh reconstruction I do not know/understand most of parameters there and the manual is (sadly) not a real help for me - I miss a clear information how each parameter works.
2. I just showed a simple example for explanation of the problem. On real buildings I'ts even worse and I wonder if it isn't a case of much earlier stages - featurematching and structurefrommotion. I noticed that feature points rarely refer to real edges (slightly better with AKAZE than SIFT) therefore point representation from SfM is "poor/not defining edges".
So.... this was my question whether anybody successfuly tested a set of settings that give optimal quality for this type of objects.

By the way - sadly 2021 doesn't work properly on my machine - it doesn't import photos at all, so it's hard to test anything on the new release.
![obraz](https://user-images.githubusercontent.com/79451289/111003996-76fcc500-8388-11eb-9a9f-7f124963298b.png)
![obraz](https://user-images.githubusercontent.com/79451289/111004141-bb886080-8388-11eb-9e07-de6da8a006c7.png)

**Reply by natowi** [reply]
I think your reconstruction is quite good, considering the shooting conditions.
As you can see, the areas where shades of the tree are have more noise. Same with the windows.
This is the  result of the trees moving in the wind, resulting in moving shades on the building surface. The reflective windows are always problematic. Best capture the images on a day with overcast sky or in such a way that there are no hard shades/reflections on the surfaces of interest.

You probably captured the images from some distance to capture all of the building. Best capture close up images if you need more details. Capturing distance and camera resolution can effect the reconstruction quality.

Here are some pointers on what parameters to change https://github.com/alicevision/meshroom/wiki/Reconstruction-parameters

**Comment by adam-bielanski** [comment]
Following previous conversations and  closed issue #977  find enclosed proposal of pipeline parameters that worked for me (as you can see below).

[Architecture-test.mg.zip](https://github.com/alicevision/meshroom/files/6339433/Architecture-test.mg.zip)

![obraz](https://user-images.githubusercontent.com/79451289/115311046-36564f80-a16f-11eb-96e3-dcda2836723a.png)
![obraz](https://user-images.githubusercontent.com/79451289/115311387-e035dc00-a16f-11eb-9dd4-41ec4b54243a.png)
